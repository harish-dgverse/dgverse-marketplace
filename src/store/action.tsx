/* eslint-disable default-param-last */
export const ACTIONS = {
  USER_LOGIN: 'USER_LOGIN',
  USER_UPDATE: 'USER_UPDATE',
  USER_LOGOUT: 'USER_LOGOUT',
  UPDATE_FROM_LS: 'UPDATE_FROM_LS',
  WALLET_PAIRED: 'WALLET_PAIRED',
};

export const userLogin = (payload: any) => ({
  type: ACTIONS.USER_LOGIN,
  payload,
});

export const userUpdate = (payload: any) => ({
  type: ACTIONS.USER_UPDATE,
  payload,
});

export const userLogout = () => ({
  type: ACTIONS.USER_LOGOUT,
});

export const updateStateFromLS = (payload: any) => ({
  type: ACTIONS.UPDATE_FROM_LS,
  payload,
});

export const walletPaired = (payload: { walletAddress: any; walletType: any; freshLogin: boolean }) => ({
  type: ACTIONS.WALLET_PAIRED,
  payload,
});
