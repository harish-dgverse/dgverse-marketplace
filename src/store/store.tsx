/* eslint-disable react/jsx-no-constructed-context-values */
import { createContext, useContext, useReducer } from 'react';

const Store = createContext<any>(null);
Store.displayName = 'Store';

export const useStore = () => useContext(Store);

export const StoreProvider = ({ children, initialState, reducer }: { children: any; initialState: any; reducer: any }) => {
  const [globalState, dispatch] = useReducer(reducer, initialState);
  return <Store.Provider value={[globalState, dispatch]}>{children}</Store.Provider>;
};
