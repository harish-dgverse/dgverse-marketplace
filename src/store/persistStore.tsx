/* eslint-disable prettier/prettier */
/* eslint-disable no-unused-vars */
import { FC, useEffect, useState } from 'react';
import { useStore } from './store';
import { userLogin, userLogout, updateStateFromLS } from './action';
import useRefreshToken from '../hooks/useRefreshToken';

const PersistStoreProvider = ({ children }: { children: any }) => {
    const [store, dispatch] = useStore();
    const { user } = store;
    const [isLoading, setIsLoading] = useState(true);
    const refresh = useRefreshToken();

    useEffect(() => {
        let isMounted = true;
        let stateFromLS = localStorage.getItem('store');
        if (stateFromLS) {
            stateFromLS = JSON.parse(stateFromLS);
            dispatch(updateStateFromLS(stateFromLS));
        }
        const verifyRefreshToken = async () => {
            try {
                await refresh();
            }
            catch (err) {
                console.error(err);
            }
            finally {
                if (isMounted) setIsLoading(false);
            }
        }

        if (!store?.accessToken)
            verifyRefreshToken();
        else setIsLoading(false);

        return () => { isMounted = false };
    }, [])

    useEffect(() => {
        console.log(`isLoading: ${isLoading}`)
        console.log(`aT: ${JSON.stringify(store?.accessToken)}`)
    }, [isLoading])
    //   return children;
    return (
        isLoading
            ? <p>Loading...</p>
            : children
    )
};

export default PersistStoreProvider;
