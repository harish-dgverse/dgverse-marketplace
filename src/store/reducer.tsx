/* eslint-disable default-param-last */
import { ACTIONS } from './action';

export const initialState: any = {
  isAuthenticated: false,
  user: null,
  accessToken: null,
  walletAddress: null,
  freshLogin: false,
  walletType: null,
};

// eslint-disable-next-line consistent-return
export const bookTableReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ACTIONS.USER_LOGIN:
      localStorage.setItem('state', JSON.stringify({ isAuthenticated: true }));
      return {
        ...state,
        isAuthenticated: true,
        user: action.payload.user,
        accessToken: action.payload.accessToken,
      };
    case ACTIONS.USER_UPDATE: {
      return {
        ...state,
        user: {
          ...state.user,
          user_name: action.payload.user_name,
          subscriptionType: action.payload.subscriptionType,
          verified: action.payload.verified,
          image_icon: action?.payload?.image_icon,
        },
      };
    }
    case ACTIONS.WALLET_PAIRED: {
      return {
        ...state,
        walletAddress: action.payload.walletAddress,
        walletType: action.payload.walletType,
        freshLogin: action.payload.freshLogin,
      };
    }
    case ACTIONS.UPDATE_FROM_LS:
      return action.payload;
    case ACTIONS.USER_LOGOUT:
      localStorage.clear();
      return initialState;
    default:
      return state;
  }
};
