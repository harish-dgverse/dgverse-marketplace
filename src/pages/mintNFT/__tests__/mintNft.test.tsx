/**
 * @jest-environment jsdom
 */
/* eslint-disable */
import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import FieldInfo from '../../../components/fieldInfo/fieldInfo';
import MintNFT from '../mintNFT';
// import axios from 'axios';
import * as store from '../../../store/store';

// const MockAdapter = require("axios-mock-adapter");
// const mock = new MockAdapter(axios);
describe('Mint NFT', () => {
    it('renders', () => {
        // const mockStore = jest.spyOn(store, 'useStore');  // spy on foo
        // mock.mockImplementation((arg: string) => [{user_id:10}]);
        // mock.onGet("/v1/user/1/assets?collectionsOwned=true").reply(200, {
        //     users: [{ id: 1, name: "John Smith" }],
        //   });
        const container = render(
            <MintNFT />,
        );
        // const container = render(
        //     <FieldInfo info='asdasd' />,
        // );
        expect(container).toMatchSnapshot();

        //   expect(queryByLabelText(/off/i)).toBeTruthy();

        //   fireEvent.click(getByLabelText(/off/i));

        //   expect(queryByLabelText(/on/i)).toBeTruthy();
    });
});
