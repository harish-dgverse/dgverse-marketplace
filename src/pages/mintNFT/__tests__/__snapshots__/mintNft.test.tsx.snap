// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mint NFT renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="asdasd"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="asdasd"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="InfoIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
